datafusion/physical-plan/src/filter.rs

```diff
@@ -371,7 +371,12 @@ impl ExecutionPlan for FilterExec {
     /// The output statistics of a filtering operation can be estimated if the
     /// predicate's selectivity value can be determined for the incoming data.
     fn statistics(&self) -> Result<Statistics> {
-        Self::statistics_helper(&self.input, self.predicate(), self.default_selectivity)
+        let stats = Self::statistics_helper(
```

Comment:
I think the global stats (`total_byte_size`) are not correct either, doesn't take into account the reduced number of columns. It should do something similar as `stats_projection` for `ProjectionExec`

Reply:
I agree the statistics calculation should be more sophisticated and I filed https://github.com/apache/datafusion/issues/13224 to track the idea

However, I am worried about trying to change how the statistics calculations work in this PR (I outlined some challenges I see in https://github.com/apache/datafusion/issues/13224)

Thus, I would like avoid doing a more substantial change in this PR (which fixes a functional bug) and we can  sort out how to improve the statistics calculations as a subsequent PR

Reply:
Makes sense